src/rt/sched/schedulerthread.h

```diff
@@ -168,6 +187,129 @@ namespace verona::rt
       }
     }
 
+    /**
+     * Mute a set of cowns. This will add the cowns to the mute set of the
+     * mutor.
+     */
+    void mute(T** cowns, size_t count)
+    {
+      auto it = mute_map.find(mutor);
+      if (it == mute_map.end())
+      {
+        auto* set = ObjectMap<T*>::create(alloc);
+        it = mute_map.insert(alloc, std::make_pair(mutor, set)).second;
+      }
+      else
+      {
+        mutor->weak_release(alloc);
+      }
+      auto& mute_set = *it.value();
+
+      for (size_t i = 0; i < count; i++)
+      {
+        auto* cown = cowns[i];
+
+        if (
+          (state == ThreadState::PreScan) || (state == ThreadState::Scan) ||
+          (state == ThreadState::AllInScan))
+        { // Messages in this cowns queue must be scanned.
+          cown->schedule();
+          continue;
+        }
+
+        Systematic::cout() << "Mute " << cown << std::endl;
+        auto bp = cown->backpressure.load(std::memory_order_relaxed);
+        bp.mute();
+        cown->backpressure.store(bp, std::memory_order_release);
+        T::acquire(cown);
+        mute_set.insert(alloc, cown);
+      }
+
+      alloc->dealloc(cowns, count * sizeof(T*));
+    }
+
+    /**
+     * Scan the mute map for cowns that may be unmuted. If `force` is true, then
+     * all mute sets in the map will be unmuted.
+     */
+    template<bool force = false>
+    void mute_map_scan()
+    {
+      // Handle unmute queue first.
+      while (true)
+      {
+        bool notify;
+        auto* msg = unmute_q.dequeue(alloc, notify);
+        if (msg == nullptr)
+          break;
+
+        if (msg->cown->backpressure.load(std::memory_order_acquire).muted())
+          unmute_cown(msg->cown);
+
+        msg->cown->weak_release(alloc);
+      }
+
+      for (auto entry = mute_map.begin(); entry != mute_map.end(); ++entry)
+      {
+        auto* m = entry.key();
+        if (
+          force ||
+          m->backpressure.load(std::memory_order_acquire).triggers_unmuting())
+        {
+          auto& mute_set = *entry.value();
+          for (auto it = mute_set.begin(); it != mute_set.end(); ++it)
+          {
+            unmute_local_cown(it.key());
+            mute_set.erase(it);
+          }
+          m->weak_release(alloc);
+          mute_map.erase(entry);
+          mute_set.dealloc(alloc);
+          alloc->dealloc<sizeof(ObjectMap<T*>)>(&mute_set);
+        }
+      }
+
+      if (mute_map.size() == 0)
+        mute_map.clear(alloc);
+    }
+
+    /**
+     * Unmute an individual cown.
```

Comment:
This is really "attempt to unmute an individual cown" rather than unmute yes?

Reply:
I'm not sure what you mean. The cown will be unmuted, either during this function or by the time a request for it to be unmuted reaches the scheduler tracking the cown.

Reply:
Right. So, this function might not unmute the cown. I don't think the comment reflects what is happening.